### ISSUE TITLE
VM image package conflicts

### ISSUE THREAD

**Issue by bazzawill** [BlackArch/blackarch#3610]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended

----------------------------------------

### Bug description

I am trying to update the OVA image and install some additional packages to make it my own. When I run try to update blackarch packages or general arch, I get package conflicts both with the suggested blackarch update and regular arch update

### Steps to reproduce

1. Download BlackArch Linux 64 bit OVA Image 
2. Import into VMWare
3. Run image and log in
4. In a terminal issue
    a. pacman -Syu --needed --overwrite='*' blackarch
or
   b. pacman -Syu
5. Respond Y to all replace x with community/x
or                              replace x with blackarch/x
(I have also tried Y to all of the later and N to the former)
Accept defaults such as option 1 out of 125 providers for tessdata

**Actual result:** 
option a. above gives the error
nodejs-lts-gallium and nodejs-lts-fermium are in conflict
option b.
gives dependency errors for java8-openjfx freerdp ffmpeg mpv

**Expected result:** 
upgrade occurs without issue

I have also investigated some of the conflicts possibly narrowing down the nodejs-lts-gallium/fermium conflict to packages
bagbak, bitwarden-cli, bloodhound, kresus, opensearch-dashboards
Some of which are not currently installed so I cannot remove them
### Screenshots

pacman -Syu --needed --overwrite='*' blackarch
![image](https://user-images.githubusercontent.com/1910311/204119022-12d1acd1-e47c-43e7-97b6-07a3dacc8fcd.png)

![image](https://user-images.githubusercontent.com/1910311/204119199-e96892bd-3a00-4f03-a3d1-c56833cb3590.png)

pacman -Syu
![image](https://user-images.githubusercontent.com/1910311/204119227-110efe43-4eef-4b44-a7a6-fa6bb43de9f9.png)

### Info for developers

**Device:** Manufacturer Model XVI (Only if you consideer needed, for example, network cards)
**GNU/Linux distribution:** (If you are using one distro that is not BlackArch, example: Manjaro, Anarchy, etc)
**Tool version:** 0.0.0

### Link to debug log
Currently I am writing this on the host machine and copy and paste is not working between my host and guest right now. 
If required I can open browser in guest and add to this issue

**Comment by dualfade** [comment]
If I remember this correctly. Let it use the gallum version ( I had no issues when doing this ). Then when you upgrade again it will go back to the standard nodejs. This is upstream and not blackarch.

again; this was quite some time ago do just recalling from memory here. 

maybe that will help.
Sincerely, 
------- Original Message -------

**Comment by bazzawill** [comment]
Thanks I have now figured out to use 
pacman -S nodejs-lts-gallum
in order to "fix" the conflict.
This does not seem intuitive to me and my initial google of the error did not offer this solution. 
Eventually I discovered it by googling package conflicts. So this is more of an user / arch issue not specific to blackarch. I have now found additional conflicts but now know how to fix them. I will close the issue once I sort through all of them. I will also include how to fix.

**Comment by dualfade** [comment]
Excellent ! Glad you got it sorted. Be good and enjoy ! ;) 
------- Original Message -------

**Comment by bazzawill** [comment]
Ok so I was getting somewhere by addressing package conflicts as they are discovered. Then I ran into pgp problems. I was aware of the issues with openssl and the jump to 3.0 . I could still not upgrade the entire system so without thinking all the way through I did 
pacman -S openssl 
I remembered after that this will render the entire system broken.
I am just going to install a new VM with the ISO. 
At this point it appears the VM is image is more effort than that.

**Comment by noraj** [comment]
> 4\. pacman -Syu --needed --overwrite='*' blackarch

Please don't do that, that will **TRY** to install all 2800+ tools. But it is impossible for multiple reasons. In a such large pool, there will always be at least one broken package but also there are legitimately several packages that have incompatible dependencies and so that will conflict.

There primary goal of the VM and full ISO is for offline install (people that don't have internet so they can download it from somewhere they have internet and bring it home but it's a very edge case, what will you do with a pentest distro without internet?). Full ISO is not better than slim or netinstall, in the contrary it will be more hassle managing conflicts without talking of the storage usage for uused tools (95%+). In the future we may remove the full ISO and ship the VM with the same amount of tools as the slim ISO because the full one is useless and just bring tons of people having dependencies issues.

> nodejs-lts-gallium and nodejs-lts-fermium are in conflict

There is nothing that can be done about that, some tools have some incompatible dependencies that lead to conflicts so you have to choose which tool you want. That's why install all 2800+ tools doesn't make sense.

But the BA website is currently misleading since it doesn't warn about all of that. 

> gives dependency errors for java8-openjfx freerdp ffmpeg mpv

If you [search in the issue tracker for java8-openjfx](https://github.com/BlackArch/blackarch/issues?q=is%3Aissue+is%3Aopen+java8-openjfx) you will see there is existing issue about it, it's deprecated but there are abandoned tools that requires deprecated dependencies and will never get fixed upstream. So either we remove them (rarely) or we keep it in the repo for people who are willing to dig and make them work at a certain cost. There will always be broken tools in the repository that why I said earlier installing all 2800+ tools can't work. In the future we may need to create an _archive_ separate repository and move abandoned / deprecated tools there to avoid this kind of issue with tools that will never get updated.

What I suggest for now if you are not comfortable with debugging all the conflicts is to either take the netinstall or slim ISO and create your own VM and only installing the tools you need.

**Comment by bazzawill** [comment]
Ok that's some good info,
I did try a fresh install using the full iso but ran into issues with pgp.
The installer ran but did not finish installing into the files into chroot.
I tried manually with pacstrap and that's where I noticed the pgp errors.
Now I am using arch VM and installing desktop and black arch on top.
That is working well. Just installing the fuzzer group now while I go to
bed.
Thanks
Barry
